Source/WebCore/dom/ElementTextDirection.cpp

```diff
@@ -0,0 +1,326 @@
+/*
```

Comment:
Can we not do this refactoring at the same time we're changing the semantics of all these code?

Reply:
The text direction code was moved back to Element.cpp